vmsdk/python/tests/tdx/test_sdk.py

```diff
@@ -0,0 +1,107 @@
+    """
+
+    RTMR_LENGTH_BY_BYTES = 48
+    """RTMR length by bytes."""
+
+    def test_get_default_algorithms(self):
+        """Test default algorithm is supported."""
+        algo = CCTrustedVmSdk.inst().get_default_algorithms()
+        assert algo is not None
+        assert algo.alg_id == TcgAlgorithmRegistry.TPM_ALG_SHA384
+
+    def test_get_measurement_count(self):
+        """Test measurement count is 4 (RTMR count)."""
+        count = CCTrustedVmSdk.inst().get_measurement_count()
+        assert count == TestCCTrustedVmSdkTdx.MEASUREMENT_COUNT
+
+    def replay_eventlog(self):
+        """Get RTMRs from event log by replay."""
+        rtmr_len = TestCCTrustedVmSdkTdx.RTMR_LENGTH_BY_BYTES
+        rtmr_cnt = TestCCTrustedVmSdkTdx.MEASUREMENT_COUNT
+        rtmrs = [bytearray(rtmr_len)] * rtmr_cnt
+        event_logs = CCTrustedVmSdk.inst().get_eventlog().event_logs
+        assert event_logs is not None
+        for event in event_logs:
+            if isinstance(event, TcgImrEvent):
+                sha384_algo = sha384()
+                sha384_algo.update(rtmrs[event.imr_index] + event.digests[0].hash)
+                rtmrs[event.imr_index] = sha384_algo.digest()
+        return rtmrs
+
+    def check_imr(self, imr_index: int, alg_id: int, rtmr: bytes):
+        """Check individual IMR.
+        Compare the 4 IMR hash with the hash derived by replay event log. They are
+        expected to be same.
+        Args:
+            imr_index: an integer specified the IMR index.
+            alg_id: an integer specified the hash algorithm.
+            rtmr: bytes of RTMR data for comparison.
+        """
+        assert 0 <= imr_index < TestCCTrustedVmSdkTdx.MEASUREMENT_COUNT
+        assert rtmr is not None
+        assert alg_id == TcgAlgorithmRegistry.TPM_ALG_SHA384
+        imr = CCTrustedVmSdk.inst().get_measurement([imr_index, alg_id])
+        assert imr is not None
+        digest_obj = imr.digest(alg_id)
+        assert digest_obj is not None
+        digest_alg_id = digest_obj.alg.alg_id
+        assert digest_alg_id == TcgAlgorithmRegistry.TPM_ALG_SHA384
+        digest_hash = digest_obj.hash
+        assert digest_hash is not None
+        assert digest_hash == rtmr, \
+            f"rtmr {rtmr.hex()} doesn't equal digest {digest_hash.hex()}"
+
+    def test_get_measurement(self):
+        """Test measurement result.
+        The test is done by compare the measurement register against the value
+        derived by replay eventlog.
+        """
+        cmdline = None
+        try:
+            with open(TestCCTrustedVmSdkTdx.PROC_CMDLINE, "r", encoding="utf-8") as proc_cmdline:
+                cmdline = proc_cmdline.readline()
+        except (PermissionError, OSError):
+            LOG.error("Need root permission to open file %s", TestCCTrustedVmSdkTdx.PROC_CMDLINE)
+            assert False
+        assert cmdline is not None
+
+        ima_policy = None
+        if "ima_hash=sha384" in cmdline:
```

Comment:
why measurement is related to ima_policy? @ruomengh

Reply:
This is my misunderstanding. I'll resolve this and update PR.